### PR TITLE
feat(website): extract `/plugins` `/tools` docs

### PR DESCRIPTION
**Pull request by thiskevinwang**
[preview](https://nomad-git-kevin-extract-plugins-hashicorp.vercel.app/plugins)

# Description

This change is related to discussions from the versioned-docs meeting on `11-04-2021`

The following plugins are versioned independently. Extracting them into a separate section on the website allows them to be displayed separately from the core nomad documentation, which we are adding versioned docs support for.
- "Task Drivers"
- "Device Plugins"
- "Autoscaling"

### CODE REVIEW

File: website/content/plugins/index.mdx [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+layout: docs
+page_title: Plugins
+description: FIXME
+---
+
+# FIXME
```
Comment (thiskevinwang) [inline website/content/plugins/index.mdx:7]: @angrycub This is the page's MDX file that I mentioned in slack.
I added basic frontmatter here, and some dummy content.

Feel free to branch-off this PR or commit directly!

**Comment by thiskevinwang** [comment]
@jescalan @BRKalow can you help sanity check this?
- redirects - https://github.com/hashicorp/nomad/pull/11584/commits/e063a0f8d507f7496a8d71f752eca1a8091a4ae6
- in-content links + old redirects https://github.com/hashicorp/nomad/pull/11584/commits/6ad1e93fbced895991631923b8fdd1fa66708666

File: website/content/plugins/devices/community/index.mdx [hunk 2]
```diff
@@ -25,5 +16,4 @@ Below is a list of community-support task drivers you can use with Nomad:
 - [USB][usb]
 
 [plugin_guide]: /docs/internals/plugins
-[nvidia]: /docs/devices/external/nvidia
-[usb]: /docs/devices/external/usb
+[usb]: /plugins/devices/nvidia
```
Comment (BRKalow) [inline website/content/plugins/devices/community/index.mdx:19]:
```suggestion
[usb]: /plugins/devices/community/usb
```

Comment (thiskevinwang) [reply website/content/plugins/devices/community/index.mdx]: Fixed - https://github.com/hashicorp/nomad/pull/11584/commits/9b8295e097bea066bb8eba3598f5248cf7317bed

File: website/data/docs-nav-data.json [hunk 3]
```diff
@@ -1498,98 +1498,98 @@
     "routes": [
       {
         "title": "Overview",
-        "path": "autoscaling"
+        "href": "/plugins/autoscaling"
```
Comment (BRKalow) [inline website/data/docs-nav-data.json:1501]: These should all be `/tools/autoscaling` 👀

Comment (thiskevinwang) [reply website/data/docs-nav-data.json]: Thanks for catching that! - Fixed 39bc48a

**Review by BRKalow — APPROVED** [review]
LGTM! Nice work

File: website/data/docs-nav-data.json (hunk 3 above)
Comment (thiskevinwang) [reply website/data/docs-nav-data.json]: 2 more stragglers - https://github.com/hashicorp/nomad/pull/11584/commits/705fa9fb006cde6a2bd02b1961e88fa10c846fe0

File: website/redirects.js [hunk 4]
```diff
@@ -1262,4 +1262,41 @@ module.exports = [
...
+  {
+    source: '/docs/drivers',
+    destination: '/plugins/drivers',
+    permanent: true,
+  },
+  {
+    source: '/docs/drivers/:splat*',
+    destination: '/plugins/drivers/:splat*',
+    permanent: true,
+  },
```
Comment (lgfa29) [inline website/redirects.js:1275]: I think this rule is breaking built-in drivers docs page, for example:
https://nomad-git-kevin-extract-plugins-hashicorp.vercel.app/docs/drivers/docker

We may need to list the ones that moved to the new section individually 🤔

**Review by lgfa29 — CHANGES_REQUESTED** [review]

File: website/redirects.js (hunk 4 above)
Comment (thiskevinwang) [reply website/redirects.js:1275]: Ah, yea I see there are a few broken ones due to this rule

Broken:
```
docs/drivers/docker
docs/drivers/exec
docs/drivers (index)
docs/drivers/java
docs/drivers/qemu
docs/drivers/raw_exec
```

Comment (thiskevinwang) [reply website/redirects.js]: Should be addressed in https://github.com/hashicorp/nomad/pull/11584/commits/9fe6f5d9812b4a0e384303b78140921cf777322d

Will verify when the preview updates

**Review by lgfa29 — APPROVED** [review]
:shipit:

**Pull request merged**